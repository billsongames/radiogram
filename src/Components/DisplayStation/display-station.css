.display-station__info {
  display: flex;
  flex-flow: column;
  align-items: center;
  margin: 10px;

}

.display-station__logo {
  border: 4px solid;
  border-radius: 12px;
  border-color: black;
  width: 200px;
  height: 200px;
  object-fit: contain;

}