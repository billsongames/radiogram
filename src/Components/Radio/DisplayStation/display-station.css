/* .display-station {
  width: 500px;
  display: grid;
  grid-template-columns: repeat(3, 300px);
} */

.display-station-container{
  width: 500px;
  display: grid;
  grid-template-columns: 100px auto;
  padding: 0px 0px 6px 0px;
}

.display-station__logo {
  border-radius: 8px;
  width: 100px;
  height: 100px;
  object-fit: contain;
}

.display-station__info {
  margin: 0px 10px 0px 10px;
  height: 100%;
  display: grid;
  grid-template-rows: auto;
}

.display-station__name {
  font-size: 18px;
  font-weight: bold;
  color: whitesmoke;
}

.display-station__hearticon {
  padding: 2px;
  height: 48px;
  width: 48px;
  object-fit: contain;
  cursor: pointer;
}

.display-station__hearticon:hover {
  scale: 1.4;
  transition-duration: 0.5s;
}

.save_info {
  display: flex;
  flex-direction: column;
  justify-content: end;
}

/* ################################################################
                                                                  #
  MEDIA QUERIES                                                   # 
                                                                  #
  ################################################################
   */

@media only screen and (max-width: 480px) {
  .display-station-container{
    width: 100%;
    display: grid;
    grid-template-columns: 60px auto;
    padding: 0px 0px 6px 0px;
  }
  
  .display-station__logo {
    border-radius: 8px;
    width: 60px;
    height: 60px;
    object-fit: contain;
  }
  
  .display-station__info {
    margin: 0px 10px 0px 10px;
    height: 60px;
    display: grid;
    grid-template-rows: auto;
  }
  
  .display-station__name {
    font-size: 12px;
    font-weight: bold;
    color: whitesmoke;
  }
  
  .display-station__hearticon {
    height: 32px;
    width: 32px;
    object-fit: contain;
    cursor: pointer;
  }
  
  .display-station__hearticon:hover {
    scale: 1.4;
    transition-duration: 0.5s;
  }
}

@media only screen and (min-width: 480px) and (max-width: 768px) {
  .display-station-container{
    width: 100%;
    display: grid;
    grid-template-columns: 60px auto;
    padding: 0px 0px 6px 0px;
  }
  
  .display-station__logo {
    border-radius: 8px;
    width: 60px;
    height: 60px;
    object-fit: contain;
  }
  
  .display-station__info {
    margin: 0px 10px 0px 10px;
    height: 60px;
    display: grid;
    grid-template-rows: auto;
  }
  
  .display-station__name {
    font-size: 14px;
    font-weight: bold;
    color: whitesmoke;
  }
  
  .display-station__hearticon {
    height: 32px;
    width: 32px;
    object-fit: contain;
    cursor: pointer;
  }
  
  .display-station__hearticon:hover {
    scale: 1.4;
    transition-duration: 0.5s;
  }  
}



@media only screen and (min-width: 768px) and (max-width: 1200px) {
  .display-station-container{
    width: 100%;
    display: grid;
    grid-template-columns: 80px auto;
    padding: 0px 0px 6px 0px;
  }
  
  .display-station__logo {
    border-radius: 8px;
    width: 80px;
    height: 80px;
    object-fit: contain;
  }
  
  .display-station__info {
    margin: 0px 10px 0px 10px;
    height: 80px;
    display: grid;
    grid-template-rows: auto;
  }
  
  .display-station__name {
    font-size: 18px;
    font-weight: bold;
    color: whitesmoke;
  }
  
  .display-station__hearticon {
    height: 36px;
    width: 36px;
    object-fit: contain;
    cursor: pointer;
  }
  
  .display-station__hearticon:hover {
    scale: 1.4;
    transition-duration: 0.5s;
  }  

}