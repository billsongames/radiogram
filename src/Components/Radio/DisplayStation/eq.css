.equaliser-container {
  display: flex;
  flex-direction: row;
  justify-content: center;
  min-height:100%;
  width:300px;
  margin: auto;
}

.clearfix {
	zoom: 1;
	&:before,
	&:after {
    content: "\0020";
    display: block;
    height: 0;
    overflow: hidden;
  }
	&:after {
    clear: both;
  }
}

.colour-bar {
  position:absolute;
  left:0;
  right:0;
  bottom:0;
  height:18px;
  background: #AEED25;
}

.equaliser-column {
  width:32px;
  float:left;
  margin:0 6px;
  padding:0;
  height:180px;
  position:relative;
  list-style-type:none;
  background:#151515;
  &:nth-child(1) {
  .colour-bar {
			animation: color-bar var(--first-anim-value) 1s ease-out alternate infinite;
    }
  }
  &:nth-child(2) {
    .colour-bar {
			animation: color-bar var(--first-anim-value) 0.5s ease-out alternate infinite;
    }
  }
  &:nth-child(3) {
    .colour-bar {
			animation: color-bar var(--first-anim-value) 1.5s ease-out alternate infinite;
    }
  }
  &:nth-child(4) {
    .colour-bar {
			animation: color-bar var(--first-anim-value) 0.25s ease-out alternate infinite;
    }
  }
  &:nth-child(5) {
    .colour-bar {
			animation: color-bar var(--first-anim-value) 2s ease-out alternate infinite;
    }
  }
}
.led-mask {
  display:block;
  margin:0;
  position:relative;
  &:nth-child(even) {
    background:#212121;
    height:6px;
  }
  &:nth-child(odd) {
    height:12px;
    background:transparent;
    box-shadow:0 0 1px rgba(0,0,0,1),
    inset 0 0 8px rgba(0,0,0,0.9);
  }
}
@keyframes color-bar {
  0% { height:18px; background:#AEED25; }
  10% { height:90px; background:#AEED25; }
  20% { height:54px; background:#AEED25; }
  30% { height:126px; background:#AEED25; }
  40% { height:162px; background:#AEED25; }
  50% { height:54px; background:#AEED25; }
  60% { height:144px; background:#AEED25; }
  70% { height:90px; background:#AEED25; }
  80% { height:126px; background:#AEED25; }
  90% { height:36px; background:#AEED25;   }
  100% { height:18px; background:#AEED25; }
}

/*input[type="checkbox"]#track-control {
	~ equaliser-container .colour-bar {
      animation:paused;
    }
  &:checked {
    ~ equaliser-container .colour-bar {
      animation:running;
    }
  }
}*/

/* background: -webkit-linear-gradient(top, #f70c0c 0%,#f70c0c 18px,#a2d822 18px,#a2d822 100%); */



