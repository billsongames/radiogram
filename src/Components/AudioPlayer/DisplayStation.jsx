import { React, useState, useRef } from "react";

import "./display-station.css"

function DisplayStation({currentStation, audioRef}) {
  return(
    <div className="display_station_info">
      <img className="display-station-logo" src={currentStation.favicon} alt={currentStation.name}/>
      <h2>{currentStation.name}</h2>
      
      <audio
        src={currentStation.urlResolved}
        ref={audioRef}
      />
    </div>
  )
}

export default DisplayStation