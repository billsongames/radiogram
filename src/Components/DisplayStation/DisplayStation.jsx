import { React, useState, useRef } from "react";
import {onPresetSaveButtonClicked} from "../Presets/Presets"
import {api_test_data} from "../../data/api_test_data"

import "./display-station.css"
import saved_preset from "../../assets/img/heart-solid.png"
import not_saved_preset from "../../assets/img/heart-regular.png"

const  DisplayStation = ({currentStation, onPresetSaveClicked}) => {

/*   let tagWords = (currentStation.tags)
  console.log(tagWords)

  for (let i=0; i<tagWords.length; i++) {
    tagWords[i] = tagWords[i][0].toUpperCase() + tagWords[i].substring(1)
  }

  const tags = tagWords.join(", ") */



  return(
    <div className="display-station">
      <div>
        <img className="display-station__logo" src={currentStation.favicon} alt={currentStation.name}/>
      </div>
      <div className="display-station__info">
        <div>
          {currentStation.name}          
        </div>
        <div>
          tags
        </div>
        <div>
          <img
            className= "display-station__hearticon"
            src={not_saved_preset}
            alt = "Station not a preset"
            onClick={onPresetSaveClicked}
            />
        </div>

      </div>
      
      
    </div>
  )
}

export default DisplayStation