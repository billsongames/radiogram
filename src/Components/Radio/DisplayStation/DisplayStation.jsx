import { React, useState, useEffect } from "react";

import { IconContext } from "react-icons";
import { BsHeartFill, BsHeart } from "react-icons/bs";

import "./display-station.css"


const  DisplayStation = ({tuned, userID, currentStation, onPresetSaveClicked, onPresetRemoveClicked, presets}) => {
  const [alreadyPreset, setAlreadyPreset] = useState(0)

  useEffect(() => {
    setAlreadyPreset(presets.findIndex(station => 
      station.id === currentStation.id))
  },[currentStation.id, presets])

  let save_info_jsx

  if (userID && tuned === true && alreadyPreset === -1) {
    save_info_jsx = (
      <IconContext.Provider value={{ color: "red", size: "36px" }}>
        <BsHeart
//          id={currentStation.id}
          className= "display-station__hearticon"
          alt = "Station not saved as preset"
          onClick={onPresetSaveClicked}
        />
      </IconContext.Provider>
    )
  } else if (userID && tuned === true && alreadyPreset >= 0) {
    save_info_jsx = (
      <IconContext.Provider value={{ color: "red", size: "36px"}}>
        <BsHeartFill
          id={currentStation.id}
          className= "display-station__hearticon"
          alt = "Station saved as preset"
          onClick={onPresetRemoveClicked}
        />
      </IconContext.Provider>
    )  
  } else if (userID && tuned === false) {
    save_info_jsx = (
      <></>
    )
  } else {
    save_info_jsx = (
      <>Log in to save presets</>
    )
  }


  return(
    <div className="display-station-container">      
      <div>
        <img className="display-station__logo" id={currentStation.id} src={currentStation.favicon} alt={currentStation.name}/>
      </div>
      <div className="display-station__info">
        <div className="display-station__name">
          {currentStation.name}          
        </div>
        <div className="save_info">
          {save_info_jsx}
        </div>
      </div>      
    </div>
  )
}

export default DisplayStation