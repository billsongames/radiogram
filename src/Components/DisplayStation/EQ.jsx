import React from "react";

import "./eq.css"

const EQ = () => {

  return(
    <div class='eq-container'>
      

 
  
  <div class="equaliser-container">
    <ol class="equaliser-column">
      <li class="colour-bar"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
    </ol>
    <ol class="equaliser-column">
      <li class="colour-bar"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
    </ol>
    <ol class="equaliser-column">
      <li class="colour-bar"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
    </ol>
    <ol class="equaliser-column">
      <li class="colour-bar"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
    </ol>
    <ol class="equaliser-column">
      <li class="colour-bar"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
      <li class="led-mask"></li>
    </ol>
  </div>
    </div>
  )
}

export default EQ