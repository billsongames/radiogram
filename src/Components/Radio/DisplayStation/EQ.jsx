import React from "react";

import "./eq.css"

const EQ = () => {

  return(
    <div className='eq-container'>
      

 
  
  <div className="equaliser-container">
    <ol className="equaliser-column">
      <li className="colour-bar"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
    </ol>
    <ol className="equaliser-column">
      <li className="colour-bar"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
    </ol>
    <ol className="equaliser-column">
      <li className="colour-bar"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
    </ol>
    <ol className="equaliser-column">
      <li className="colour-bar"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
    </ol>
    <ol className="equaliser-column">
      <li className="colour-bar"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
      <li className="led-mask"></li>
    </ol>
  </div>
    </div>
  )
}

export default EQ