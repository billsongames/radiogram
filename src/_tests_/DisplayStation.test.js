import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";

import DisplayStation from "../Components/Radio/DisplayStation/DisplayStation";

describe("DisplayStation", () => {
  const validProps = {
    currentStation : [
      {
        id: "test id",
        name: "test name",
        favicon: "test_favicon.jpg",
        urlResolved: "url 1"
      }
    ],
    onPresetSaveClicked: jest.fn(),
    onPresetRemoveClicked: jest.fn(),
    presets: []  
  }

  test("renders correctly", () => {
    const { asFragment } = render(<DisplayStation
      currentStation = {validProps.currentStation}
      onPresetSaveClicked = {validProps.onPresetSaveClicked}
      onPresetRemoveClicked = {validProps.onPresetRemoveClicked}
      presets = {validProps.presets}
    />)
    
    expect(asFragment()).toMatchSnapshot();
  })
})

describe("Save preset function", () => {
  const validProps = {
    userID: "test userID",
    tuned: true,
    currentStation : [
      {
        id: "test id",
        name: "test name",
        favicon: "test_favicon.jpg"
      }
    ],
    onPresetSaveClicked: jest.fn(),
    onPresetRemoveClicked: jest.fn(),
  }

  test("preset save clicked calls correct function", async () => {
    render(<DisplayStation
      userID = {validProps.userID}
      tuned = {validProps.tuned}
      currentStation = {validProps.currentStation}
      onPresetSaveClicked = {validProps.onPresetSaveClicked}
      onPresetRemoveClicked = {validProps.onPresetRemoveClicked}
      presets = {[]}
    />)

    const button = screen.getByTestId("save_preset")
    await fireEvent.click(button)

    expect(validProps.onPresetSaveClicked).toHaveBeenCalledTimes(1)    
  })

})

describe("Remove preset function", () => {
  const validProps = {
    userID: "test userID",
    tuned: true,
    currentStation : [
      {
        id: "test id",
        name: "test name",
        favicon: "test_favicon.jpg"
      }
    ],
    onPresetSaveClicked: jest.fn(),
    onPresetRemoveClicked: jest.fn(),
  }

  xtest("remove save clicked calls correct function", async () => {
    const presets = [
      {
        id: "test id",
        name: "test name",
        favicon: "test_favicon.jpg"
      }
    ]

    render(<DisplayStation
      userID = {validProps.userID}
      tuned = {validProps.tuned}
      currentStation = {validProps.currentStation}
      onPresetSaveClicked = {validProps.onPresetSaveClicked}
      onPresetRemoveClicked = {validProps.onPresetRemoveClicked}
      presets = {presets}

    />)

    const button = screen.getByTestId("remove_preset")
    await fireEvent.click(button)

    expect(validProps.onPresetRemoveClicked).toHaveBeenCalledTimes(1)    
  }) 


})


