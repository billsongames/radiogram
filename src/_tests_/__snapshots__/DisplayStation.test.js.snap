// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 DisplayStation renders correctly 1`] = `
<DocumentFragment>
  <div
    class="display-station-container"
  >
    <div>
      <img
        class="display-station__logo"
      />
    </div>
    <div
      class="display-station__info"
    >
      <div
        class="display-station__name"
      />
      <div
        class="save_info"
      >
        Log in to save presets
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DisplayStation renders correctly 1`] = `
<DocumentFragment>
  <div
    class="display-station-container"
  >
    <div>
      <img
        class="display-station__logo"
      />
    </div>
    <div
      class="display-station__info"
    >
      <div
        class="display-station__name"
      />
      <div
        class="save_info"
      >
        Log in to save presets
      </div>
    </div>
  </div>
</DocumentFragment>
`;
